app/components/ToggleDisplay.js

```diff
@@ -80,28 +79,17 @@ export default class ToggleDisplay extends Component {
 
   runHelpers = () => {
     if (this.state.helpers) {
-      let shouldHide = false;
-      let obj = {};
-
       chrome.storage.local.get('display', result => {
-        this.state.helpers.map(helper => {
+        this.state.helpers.forEach(helper => {
           if (helper.type === 'hide') {
             if (!{}.hasOwnProperty.call(result.display, helper.relation)) {
-              return helper;
+              return;
             }
             const relation = result.display[helper.relation].visible;
-            shouldHide = this.state.visible + relation;
-            obj = helper;
-            obj.display = true;
+            helper.display = (this.state.visible + relation) !== 0;
+            this.sendHelperToDOM(helper);
           }
```

Comment:
Just wanted to make sure we both agree on this new implementation being what we want now.

So right now it takes each helper and for each of them:
 - Checks that the type is 'hide' which is the only type we support
 - Checks that the helper has a relation (since we check that it has a relation, should we also check it has a selector since we aren't trusting the config file in regards to relation? Though I see it wouldn't throw an error not having a selector while it would missing the relation. An incorrect relation would still result in an error though as undefined.visible would throw an error)
 - We reassign the helper's display property. I guess this actually mutates the Config.js file that's in memory (of course not the one on the harddrive). So this is I guess an example of hidden stuff going on that the airbnb style rule tries to prevent. I guess it won't hurt anything though, as any `helper.display` is always assigned before it's sent on though. I guess one could also think of edge cases such as running this render fast twice in a row and it could mutate the helper object again while it's being sent. I don't think in effect that it would cause any problems though.

Is this the hoped functionality of this function?